react-router/react-router-tests.tsx

```diff
@@ -12,6 +12,16 @@ import { browserHistory, hashHistory, Router, Route, IndexRoute, Link } from "re
 
 class Master extends React.Component<React.Props<{}>, {}> {
 
```

Comment:
Add this here instead of the type casting that you're doing below:

```ts
static contextTypes: React.ValidationMap<any> = {
    router: React.PropTypes.object
};

context: {
    router: ReactRouter.RouterOnContext
};